components/registry/org.wso2.carbon.registry.extensions/src/main/java/org/wso2/carbon/registry/extensions/handlers/ZipWSDLMediaTypeHandler.java

```diff
@@ -244,6 +245,14 @@ public void put(RequestContext requestContext) throws RegistryException {
                             entry = zs.getNextEntry();
                             while (entry != null) {
                                 String entryName = entry.getName();
+                                String destPath = tempFile.getAbsolutePath().substring(0, tempFile.getAbsolutePath().length()
+                                        - archiveExtension.length());
+                                String canonicalEntryPath = new File(destPath + File.separator
+                                        + entryName).getCanonicalPath();
+                                String canonicalDirPath = new File(destPath).getCanonicalPath();
+                                if (!canonicalEntryPath.startsWith(canonicalDirPath)) {
+                                    throw new IOException("Entry is outside of the target directory: " + entry.getName());
```

Comment:
Shall we throw a RegistryException as it's already defined in the method signature?